lib/fs.js

```diff
@@ -826,7 +825,11 @@ function ftruncate(fd, len = 0, callback) {
   }
   validateInt32(fd, 'fd', 0);
   validateInteger(len, 'len');
-  len = MathMax(0, len);
+
+  if (len < 0) {
+    throw new ERR_INVALID_ARG_VALUE('len', len);
+  }
+
```

Comment:
To be more consistent, I was thinking if this should be wrapped in a validation function like this:
```suggestion
validateInteger(len, 'len', 0, 0xffffffff000);
```
`0xffffffff000` seems to be the max limit that doesn't cause an `EFBIG` for `ftruncate` on my 32 bit Linux running Node v12.16.3. Is it the same for other configurations too?

Reply:
> 0xffffffff000 seems to be the max limit that doesn't cause an EFBIG for ftruncate on my 32 bit Linux running Node v12.16.3. Is it the same for other configurations too?

This is most likely going to be very OS- and architecture-dependent. I would not hardcode a fixed maximum value here.

Reply:
@addaleax okay, is there any other way we can handle the case where `len` hits the upper limit? Or, is it better if we just leave the way it is currently?

Reply:
@raisinten We do what we currently do, i.e. wait for the operating system to give us an error if the value is too large.